### PR TITLE
linked in provider options and profile update

### LINKED ISSUE #8316: `Sign In with LinkedIn` is deprecated, use `Sign In with LinkedIn using OpenID Connect` instead
### Provider type

LinkedIn

### Environment

System:
  OS: macOS 13.4.1
  CPU: (8) arm64 Apple M1
  Memory: 82.78 MB / 8.00 GB
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
  Yarn: 1.22.19 - ~/.nvm/versions/node/v16.17.0/bin/yarn
  npm: 9.6.5 - ~/.nvm/versions/node/v16.17.0/bin/npm
  pnpm: 7.5.2 - ~/.nvm/versions/node/v16.17.0/bin/pnpm
Browsers:
  Chrome: 115.0.5790.170
  Safari: 16.5.2

### Reproduction URL

https://github.com/setren/linkedin-provider

### Describe the issue

Linkedin changed its authentication method from `Sign in with LinkedIn` to `Sign in with LinkedIn using OpenID Connect`, since then the new app registered recently had this error 
> https://next-auth.js.org/errors#oauth_callback_handler_error unauthorized_scope_error {
  error: {
    message: 'unauthorized_scope_error',
    stack: 'Error: unauthorized_scope_error\n' +
      '    at oAuthCallback (webpack-internal:///(api)/./node_modules/next-auth/core/lib/oauth/callback.js:39:19)\n' +
      '    at Object.callback (webpack-internal:///(api)/./node_modules/next-auth/core/routes/callback.js:52:39)\n' +
      '    at AuthHandler (webpack-internal:///(api)/./node_modules/next-auth/core/index.js:208:41)\n' +
      '    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n' +
      '    at async NextAuthApiHandler (webpack-internal:///(api)/./node_modules/next-auth/next/index.js:22:19)\n' +
      '    at async NextAuth._args$ (webpack-internal:///(api)/./node_modules/next-auth/next/index.js:107:14)',
    name: 'Error'
  },
  error_description: 'Scope &quot;r_liteprofile&quot; is not authorized for your application',
  providerId: 'linkedin',
  message: 'unauthorized_scope_error'
}

It seems that scope variable changed from `"r_liteprofile r_emailaddress"` to `"profile email"` https://learn.microsoft.com/en-us/linkedin/shared/authentication/getting-access#open-permissions-consumer. I have tried to add scope manually ` LinkedinProvider({
      clientId: process.env.LINKEDIN_CLIENT_ID!,
      clientSecret: process.env.LINKEDIN_CLIENT_SECRET!,
      authorization: {
        params: {
          scope: "profile email",
        },
      },
    }),` but it gives this error
> [next-auth][error][OAUTH_CALLBACK_ERROR] 
https://next-auth.js.org/errors#oauth_callback_error expected 200 OK, got: 403 Forbidden {
  error: OPError: expected 200 OK, got: 403 Forbidden
      at processResponse (/Users/setren/Documents/linkedin-auth/node_modules/openid-client/lib/helpers/process_response.js:41:11)
      at Client.userinfo (/Users/setren/Documents/linkedin-auth/node_modules/openid-client/lib/client.js:1237:18)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async oAuthCallback (webpack-internal:///(api)/./node_modules/next-auth/core/lib/oauth/callback.js:131:17)
      at async Object.callback (webpack-internal:///(api)/./node_modules/next-auth/core/routes/callback.js:52:11)
      at async AuthHandler (webpack-internal:///(api)/./node_modules/next-auth/core/index.js:208:28)
      at async NextAuthApiHandler (webpack-internal:///(api)/./node_modules/next-auth/next/index.js:22:19)
      at async NextAuth._args$ (webpack-internal:///(api)/./node_modules/next-auth/next/index.js:107:14) {
    name: 'OAuthCallbackError',
    code: undefined
  },
  providerId: 'linkedin',
  message: 'expected 200 OK, got: 403 Forbidden'
}

Similar problem using firebase https://stackoverflow.com/questions/76890650/sign-in-with-linkedin-using-openid-connect-with-firebase-auth-or-gcp-identity-pl

### How to reproduce

-Clone this repo https://github.com/nextauthjs/next-auth-example
-Add linkedin provider 

### Expected behavior

The user is available to sign in.

### PR DESCRIPTION
**Pull request by HugeLetters**
## ☕️ Reasoning

It seems LinkedIn has updated its authentication procedure - this PR updates the provider to reflect those changes. 
LinkedIn renamed required scopes(permissions) and also changed the shape of profile data.

## 🧢 Checklist

- [x] Documentation - not needed
- [ ] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes: #8316 

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by joemulberry** [comment]
would love this to be solved

**Comment by ss-zheng** [comment]
OMG I was stucked for a night in order to enable linkedin auth, and found this PR this morning. Please merge this appreciated it!

File: packages/core/src/providers/linkedin.ts [hunk 1]
```diff
@@ -84,33 +72,20 @@ export default function LinkedIn<P extends LinkedInProfile>(
     id: "linkedin",
     name: "LinkedIn",
     type: "oauth",
```
Comment (ThangHuuVu) [inline packages/core/src/providers/linkedin.ts:74]: you can actually remove most of the configurations:
- type should be `oidc`
- `authorization` can be removed 
- `profile` can be removed

You can take a look at the Okta provider for example.

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]
Thanks for the contribution, I left some comments. 🙏

File: packages/core/src/providers/linkedin.ts [hunk 2]
```diff
@@ -83,36 +71,12 @@ export default function LinkedIn<P extends LinkedInProfile>(
...
     name: "LinkedIn",
-    type: "oauth",
-    authorization: {
-      url: "https://www.linkedin.com/oauth/v2/authorization",
-      params: { scope: "r_liteprofile r_emailaddress" },
-    },
-    token: "https://www.linkedin.com/oauth/v2/accessToken",
+    type: "oidc",
+    wellKnown:
+      "https://www.linkedin.com/oauth/.well-known/openid-configuration",
```
Comment (ThangHuuVu) [inline packages/core/src/providers/linkedin.ts:76]:
```suggestion
```

**Review by ThangHuuVu — APPROVED** [review]

**Comment by AnatoleT** [comment]
Would love to see this solved too :)

File: packages/core/src/providers/linkedin.ts (hunk 2 above)
Comment (paramaggarwal) [reply packages/core/src/providers/linkedin.ts]: @ThangHuuVu @HugeLetters 

This suggestion was wrong, the default wellKnown of `https://www.linkedin.com/.well-known/openid-configuration` does not work!

Screenshot 2024-03-09 at 6 44 38 AM

File: packages/core/src/providers/linkedin.ts [hunk 3]
```diff
@@ -72,8 +72,6 @@ export default function LinkedIn<P extends LinkedInProfile>(
     id: "linkedin",
     name: "LinkedIn",
     type: "oidc",
-    wellKnown:
-      "https://www.linkedin.com/oauth/.well-known/openid-configuration",
```
Comment (paramaggarwal) [inline packages/core/src/providers/linkedin.ts]: Just before merging you broke the integration @ThangHuuVu

**Pull request merged**